### PR TITLE
Feat/migrator

### PR DESCRIPTION
**Pull request by mtt-artis**
Hey

This PR introduce de `protocol` key on the `FileMigrationProvider` 
This allows window users to add a protocol if they need to 

https://github.com/kysely-org/kysely/pull/430

Thanks again for this amazing library

### CODE REVIEW

**Comment by mtt-artis** [comment]
https://github.com/kysely-org/kysely/pull/430#issuecomment-1691251528

File: src/migration/file-migration-provider.ts [hunk 1]
```diff
@@ -31,11 +31,13 @@ export class FileMigrationProvider implements MigrationProvider {
     for (const fileName of files) {
       if (
         fileName.endsWith('.js') ||
+        fileName.endsWith('.cjs') ||
         (fileName.endsWith('.ts') && !fileName.endsWith('.d.ts')) ||
         fileName.endsWith('.mjs') ||
         (fileName.endsWith('.mts') && !fileName.endsWith('.d.mts'))
       ) {
         const migration = await import(
+            this.#props.protocol +
```
Comment (koskimas) [inline src/migration/file-migration-provider.ts:40]: This is now `undefined` by default and will break everything.

Comment (mtt-artis) [reply src/migration/file-migration-provider.ts]: sorry 🙏 
did this qwickly from work on the github vscode web interface 
this should fix it

i would prefer rather this solution than this ?

```ts
const isEsm = Boolean(typeof module !== 'undefined' && typeof module.imports !== 'undefined') 
this.#props.path.join(
  isEsm ? "file://" : "",
  this.#props.migrationFolder,
  fileName
)
```

**Comment by mtt-artis** [comment]
Thanks !

**Pull request closed without merging**